### PR TITLE
statistics: use DatumMapCache to remove duplicate code

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: ref #46158

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/cmsketch_util.go
Comment (hi-rustin) [inline statistics/cmsketch_util.go:1]: Better add some unit tests for this util. And maybe we can rename it to datum_map_cache. The `util` package always grows up too wild and crazy.

**Review by hi-rustin — APPROVED** [review]
👍🏿

File: statistics/cmsketch_util.go
Comment (hawkingrei) [reply statistics/cmsketch_util.go:1]: Recently, we will start to refactor the statistics package by splitting components into more package.

Comment (hawkingrei) [reply statistics/cmsketch_util.go:1]: Use the old test to verify it.

File: statistics/cmsketch_util.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	"time"
+
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/codec"
+	"github.com/pingcap/tidb/util/hack"
+)
+
+type datumMapCache struct {
+	datumMap map[hack.MutableString]types.Datum
+}
```
Comment (time-and-fate) [inline statistics/cmsketch_util.go:27]: Why do we need an extra `struct`?
I think it's the same as following:
```go
type datumMap map[hack.MutableString]types.Datum
```

Comment (time-and-fate) [reply statistics/cmsketch_util.go:27]: Personally, I think there's no need to add a new type just to wrap a simple `map` with `Get` and `Put`.
There are only two places using this, and it's not complicated.

**Review by time-and-fate — APPROVED** [review]
Others LGTM. Feel free to /unhold it.

File: statistics/cmsketch_util.go (hunk 1 above)
Comment (hawkingrei) [reply statistics/cmsketch_util.go:27]: Actually, both definitions of ```datumMapCache``` are acceptable, but I just prefer the current form.

**Pull request merged**